File: Java/0014_longest_common_prefix/0014_longest_common_prefix.java
import java.utils.*;
class Solution {
    public String longestCommonPrefix(String[] strs) {

        if (strs.length == 0) return "";

        String prefix = strs[0];

        for (int i = 1; i < strs.length; i++) {
            while (strs[i].indexOf(prefix)) {
                prefix = prefix.substring(0, str.length() - 1);
                System.out.println(prefix);
            }
        }
        return prefix;
    } 

    public static void main(String[] args) {
        Solution sol = new Solution();
        String [] str = {"flower", "flow", "flight"};
        System.out.println(sol.longestCommonPrefix(str));
    }
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
let longestCommonPrefix = function(strs) {
    if (strs.length === 0) return ""; // return empty string if there's no element

    let prefix = strs[0] // sets the initial prefix

    for (let i = 1; i < strs.length; i++) {
        while (strs[i].indexOf(prefix) !== 0) { // check if it's not common
            prefix = prefix.substring(0, prefix.length - 1) // cut the last character of prefix
        }
    }

    return prefix;
};